keras/models.py

```diff
@@ -179,13 +187,18 @@ def get_json_type(obj):
                         else:
                             param_dset[:] = val
         f.flush()
+    finally:
+        if opened_new_file:
+            f.close()
```

Comment:
In the original exit method, `close()` is only done if `f.id` is set. https://github.com/h5py/h5py/blob/4b5a901fb297f6ae5a51ff992aa8a626a7f3c3a2/h5py/_hl/files.py#L359

Is this something important?

Reply:
To the best of my understanding given the `h5py` code and documentation, `f.id` is simply a check for whether the file is still open or not. This is based on the `close()` method also checking the validity of `f.id`: https://github.com/h5py/h5py/blob/4b5a901fb297f6ae5a51ff992aa8a626a7f3c3a2/h5py/_hl/files.py#L325

As I'm going to implement your suggestion and only set `opened_new_file = True` after the file is already opened, I believe that checking `f.id` as well is unnecessary when we've already checked `opened_new_file` right before.